File: android/src/main/java/com/getcapacitor/community/classes/options/SendOptions.java
package com.getcapacitor.community.classes.options;

import static com.getcapacitor.community.BridgefyHelper.makeUUID;

import android.util.Base64;
import android.util.Pair;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.getcapacitor.JSObject;
import com.getcapacitor.PluginCall;
import java.util.UUID;

public class SendOptions {

    @Nullable
    private byte[] data;

    @Nullable
    private Pair<String, UUID> transmissionMode;

    public SendOptions(PluginCall call) {
        @Nullable
        String data = call.getString("data");
        this.setData(data);

        @Nullable
        JSObject transmissionMode = call.getObject("transmissionMode");
        if (transmissionMode != null) {
            String type = transmissionMode.getString("type");
            String uuid = transmissionMode.getString("uuid");
            if (type != null && !type.isEmpty() && uuid != null && !uuid.isEmpty()) {
                this.setTransmissionMode(type, makeUUID(uuid));
            }
        }
    }

    private void setData(@Nullable String data) {
        this.data = (data == null || data.isEmpty()) ? null : Base64.decode(data, Base64.NO_WRAP);
    }

    private void setTransmissionMode(@NonNull String type, @Nullable UUID uuid) {
        this.transmissionMode = new Pair<>(type, uuid);
    }

    @Nullable
    public byte[] getData() {
        return data;
    }

    @Nullable
    public Pair<String, UUID> getTransmissionMode() {
        return transmissionMode;
    }
}

File: ios/Sources/BridgefyPlugin/Classes/Options/SendOptions.swift
import Foundation
import Capacitor

@objc public class SendOptions: NSObject {
    private var data: Data?
    private var transmissionMode: (String, UUID)?

    init(_ call: CAPPluginCall) {
        super.init()

        if let data = call.getString("data") {
            self.setData(data)
        }

        if let transmissionMode = call.getObject("transmissionMode") {
            self.setTransmissionMode(transmissionMode["type"] as? String, transmissionMode["uuid"] as? String)
        }
    }

    func setData(_ data: String?) {
        self.data = Helper.makeData(data)
    }
    func setTransmissionMode(_ type: String?, _ uuid: String?) {
        if let type = type, let uuid = Helper.makeUUID(uuid) {
            self.transmissionMode = (type, uuid)
        }
    }

    func getData() -> Data? {
        return self.data
    }
    func getTransmissionMode() -> (String, UUID)? {
        return self.transmissionMode
    }
}
